homeassistant/components/gree/switch.py

```diff
@@ -40,126 +73,31 @@ def init_device(coordinator):
-        self.async_write_ha_state()
-
-
-class GreeQuietModeSwitchEntity(GreeEntity, SwitchEntity):
-    """Representation of the quiet mode state of the device."""
-
-    def __init__(self, coordinator):
-        """Initialize the Gree device."""
-        super().__init__(coordinator, "Quiet")
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return SwitchDeviceClass.SWITCH
-
-    @property
-    def is_on(self) -> bool:
-        """Return if the state is turned on."""
-        return self.coordinator.device.quiet
-
-    async def async_turn_on(self, **kwargs: Any) -> None:
-        """Turn the entity on."""
-        self.coordinator.device.quiet = True
-        await self.coordinator.push_state_update()
-        self.async_write_ha_state()
-
-    async def async_turn_off(self, **kwargs: Any) -> None:
-        """Turn the entity off."""
-        self.coordinator.device.quiet = False
-        await self.coordinator.push_state_update()
-        self.async_write_ha_state()
-
+class GreeSwitch(GreeSwitchEntityDescription, GreeEntity):
+    """Generic Gree entity to use with a modern style configuration."""
 
-class GreeFreshAirSwitchEntity(GreeEntity, SwitchEntity):
-    """Representation of the fresh air mode state of the device."""
+    entity_description: GreeSwitchEntityDescription
 
-    def __init__(self, coordinator):
+    def __init__(self, coordinator, description: GreeSwitchEntityDescription) -> None:
         """Initialize the Gree device."""
-        super().__init__(coordinator, "Fresh Air")
+        self.entity_description = description
 
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return SwitchDeviceClass.SWITCH
-
-    @property
-    def is_on(self) -> bool:
-        """Return if the state is turned on."""
-        return self.coordinator.device.fresh_air
-
-    async def async_turn_on(self, **kwargs: Any) -> None:
-        """Turn the entity on."""
-        self.coordinator.device.fresh_air = True
-        await self.coordinator.push_state_update()
-        self.async_write_ha_state()
-
-    async def async_turn_off(self, **kwargs: Any) -> None:
-        """Turn the entity off."""
-        self.coordinator.device.fresh_air = False
-        await self.coordinator.push_state_update()
-        self.async_write_ha_state()
-
-
-class GreeXFanSwitchEntity(GreeEntity, SwitchEntity):
-    """Representation of the extra fan mode state of the device."""
-
-    def __init__(self, coordinator):
-        """Initialize the Gree device."""
-        super().__init__(coordinator, "XFan")
-
-    @property
-    def device_class(self):
-        """Return the class of this device, from component DEVICE_CLASSES."""
-        return SwitchDeviceClass.SWITCH
+        if name := self.entity_description.name:
+            super().__init__(coordinator, name)
```

Comment:
Why do you have a condition here?

Reply:
Because MyPy was throwing an optional error if I didn't 

When the code was:

```python
super().__init__(coordinator, self.entity_description.name)
```
>homeassistant/components/gree/switch.py:86: error: Argument 2 to "__init__" of "GreeSwitchEntityDescription" has incompatible type "Optional[str]"; expected "str"  [arg-type]

So my assumption was that field is ALWAYS set so the code block as-is would always run.

Eventually there is more refactoring that could be done but not having written the integration I didn't want to completely change everything...

Reply:
It is expected to always be there.
I think it would be better to use `cast` or `assert` instead.
```suggestion
        super().__init__(coordinator, cast(str, description.name))
```

Reply:
Or make it optional in the base class